modules/pagerduty/pagerduty.go

```diff
@@ -0,0 +1,188 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Contributor: Miles Crabill <mcrabill@mozilla.com>
+
+package pagerduty
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	pagerduty "github.com/PagerDuty/go-pagerduty"
+
+	"go.mozilla.org/userplex/modules"
+)
+
+func init() {
+	modules.Register("pagerduty", new(module))
+}
+
+type module struct{}
+
+func (m *module) NewRun(c modules.Configuration) modules.Runner {
+	r := new(run)
+	r.Conf = c
+	return r
+}
+
+type run struct {
+	Conf modules.Configuration
+	p    parameters
+	c    credentials
+
+	pagerdutyEmailToLdapEmail  map[string]string
+	ldapEmailToPagerdutyEmails map[string][]string
+}
+
+type parameters struct {
+	Subdomain string
+}
+
+type credentials struct {
+	APIKey string
+}
+
+func (r *run) Run() (err error) {
+	err = r.Conf.GetParameters(&r.p)
+	if err != nil {
+		return
+	}
+	err = r.Conf.GetCredentials(&r.c)
+	if err != nil {
+		return
+	}
+
+	r.buildLdapMapping()
+	ldapers := r.getLdaperEmails()
+
+	client := pagerduty.NewClient(r.c.APIKey)
+
+	var allUsers []pagerduty.User
+	offset := uint(0)
```

Comment:
interesting way to declare a uint. I usually just do `var offset uint` since Go guarantees initialization to zero. That said, I don't have a problem with this.